tests/paddle/test_paddle_model_export.py

```diff
@@ -295,3 +295,222 @@ def test_model_log_without_specified_conda_env_uses_default_env_with_expected_de
+    model = pd_model_built_in_high_level_api.model
+    test_dataset = pd_model_built_in_high_level_api.inference_dataframe
+    with TempDir(chdr=True, remove_on_exit=True) as tmp:
+        for should_start_run in [False, True]:
+            try:
+                mlflow.set_tracking_uri("test")
+                if should_start_run:
+                    mlflow.start_run()
+
+                artifact_path = "model"
+                conda_env = os.path.join(tmp.path(), "conda_env.yaml")
+                _mlflow_conda_env(conda_env, additional_pip_deps=["paddle"])
+
+                mlflow.paddle.log_model(
+                    pd_model=model, artifact_path=artifact_path, conda_env=conda_env
+                )
+                model_uri = "runs:/{run_id}/{artifact_path}".format(
+                    run_id=mlflow.active_run().info.run_id, artifact_path=artifact_path
+                )
+
+                reloaded_pd_model = mlflow.paddle.load_model(model_uri=model_uri)
+
+                low_level_test_dataset = [x[0] for x in test_dataset]
+
+                np.testing.assert_array_almost_equal(
+                    np.array(model.predict(test_dataset)).squeeze(),
+                    np.array(reloaded_pd_model(np.array(low_level_test_dataset))).squeeze(),
+                )
+
+                model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+                model_config = Model.load(os.path.join(model_path, "MLmodel"))
+                assert pyfunc.FLAVOR_NAME in model_config.flavors
+                assert pyfunc.ENV in model_config.flavors[pyfunc.FLAVOR_NAME]
+                env_path = model_config.flavors[pyfunc.FLAVOR_NAME][pyfunc.ENV]
+                assert os.path.exists(os.path.join(model_path, env_path))
+
+            finally:
+                mlflow.end_run()
+                mlflow.set_tracking_uri(old_uri)
+
+
+@pytest.fixture
+def model_retrain_path(tmpdir):
+    return os.path.join(str(tmpdir), "model_retrain")
+
+
+@pytest.mark.large
+def test_model_retrain_built_in_high_level_api(
+    pd_model_built_in_high_level_api, model_path, model_retrain_path
+):
+    model = pd_model_built_in_high_level_api.model
+    mlflow.paddle.save_model(pd_model=model, path=model_path, training=True)
+
+    training_dataset, test_dataset = get_dataset_built_in_high_level_api()
+
+    model_retrain = paddle.Model(UCIHousing())
+    model_retrain = mlflow.paddle.load_model(model_uri=model_path, model=model_retrain)
+    optim = paddle.optimizer.Adam(learning_rate=0.015, parameters=model.parameters())
+    model_retrain.prepare(optim, paddle.nn.MSELoss())
+
+    model_retrain.fit(training_dataset, epochs=6, batch_size=8, verbose=1)
+
+    mlflow.paddle.save_model(pd_model=model_retrain, path=model_retrain_path, training=False)
+
+    with pytest.raises(TypeError, match="This model can't be loaded"):
+        mlflow.paddle.load_model(model_uri=model_retrain_path, model=model_retrain)
+
+    error_model = str(0)
```

Comment:
`str(0)` is fine but I'd simply use `0`.

Reply:
Year! 0 is better. I found a potential bug because of this!